unstable/propertyvalue/propertyvalue_test.go

```diff
@@ -30,3 +32,40 @@ func TestRemoveSecrets(t *testing.T) {
 		}
 	})
 }
+
+func TestIsNilArray(t *testing.T) {
+	t.Parallel()
+
+	require.True(t, isNilArray(resource.PropertyValue{V: []resource.PropertyValue(nil)}))
```

Comment:
I am wondering now, why are these things valid values in Pulumi?  

Could we assert against their existence or normalize them to a normal semantic form?

That is, I understand that in TF type array may have a distinct `nil` value, but it is surprising to me that in either Pulumi or TF space we would want to make a distinction between `nil` and a special `nil array`.  I would expect that at this layer we want to have only one `nil`. 

So whichever code constructed this value to represent a nil:

```
resource.PropertyValue{V: []resource.PropertyValue(nil)}
```

Should have constructed this value instead which canonically represents `nil`:

```
resource.PropertyValue{}
```

?

Reply:
Yeah, we shouldn't be making these in the first place. I've opened https://github.com/pulumi/pulumi-terraform-bridge/issues/2662 to follow up here.

However, I still think that Transform shouldn't be changing values as well, so I think this PR is still valid. Here it changes a `nil` to `empty` which are semantically different.

Reply:
Yeah I'm ok with the follow up.